src/node.cpp

```diff
@@ -103,10 +105,23 @@ void publish_scan(ros::Publisher *pub,
     pub->publish(scan_msg);
 }
 
-bool checkRPLIDARHealth(RPlidarDriver * drv)
+void publish_diag(const std::string &serial_port, int8_t level, const std::string &message, ros::Publisher &diag_pub) 
+{
+    diagnostic_msgs::DiagnosticArray darray;
+    diagnostic_msgs::DiagnosticStatus status;
+    status.level = level;
+    status.name = "rplidar driver";
+    status.hardware_id = serial_port;
+    status.message = message;
+    darray.status.push_back(status);
+    diag_pub.publish(darray);
```

Comment:
The driver only checks once at startup and then dies if it fails. I suppose we could check periodically after startup, but I didn't want to potentially* impact its performance & reliability by pinging for healthcodes so often esp. since I haven't seen it fail while operating. Additionally, I added a frequency monitor on the `/scan` topic too, so if it stops publishing, we'll know about ithat (which has happened when the rplidar decides to stop spinning).

\* potentially since I don't know how the rplidar driver works and didn't want to topple over a potentially unstable house of cards